Islandora-Devops/islandora-starter-site
# Solr hOCR search

## What does this pull request do?

Adds configurations to support a search within a paged content object's children for text found in the page object's hOCR.

## How should this be tested?

From a clean ISLE-DC instance:

Add this line to build/docker-compose/docker-compose.drupal.yml in the environment: section:

```yaml
SOLR_HOCR_PLUGIN_PATH: '/opt/solr/server/solr/contrib/ocrhighlighting/lib'
```

Then from the isle-dc folder, clone this repository with the solr-hocr branch:

```shell
git clone git@github.com:Islandora-Devops/islandora-starter-site.git --branch=solr-hocr
```

Run 'make solr_dev' and wait for it to complete.

Then after logging in as admin:

1. Create a Repository Item node with model 'Paged Content
2. Click on 'Children' and 'Batch Upload Children'.
3. Select Repository Item, Page, File, and Original File on the batch upload page and click Next.
4. Upload one or more TIFF or JP2 files and click Finish.
5. Go back to the Paged Content node, you should see the Mirador viewer and text should be selectable.
6. Click the show sidebar button in Mirador and then click the Search button.
7. In the search box, enter a term you know is in the text of an image you've uploaded.

You should see the search result in the viewer's sidebar, clicking on it should take you to the highlighted result in the viewer.

<img width="1512" alt="image" src="https://github.com/user-attachments/assets/d94d8840-950e-4d56-85fa-70c54fa7e9ac">
